addons/binding/org.openhab.binding.bosesoundtouch10/src/main/java/org/openhab/binding/bosesoundtouch/handler/BoseSoundTouchHandler.java

```diff
@@ -0,0 +1,397 @@
+            this.value = value;
+        }
+
+        public String getValue() {
+            return value;
+        }
+    }
+
+    public enum BSTKeys {
+        PLAY,
+        PAUSE,
+        POWER,
+        PRESET_1,
+        PRESET_2,
+        PRESET_3,
+        PRESET_4,
+        PRESET_5,
+        PRESET_6,
+    }
+
+    public BoseSoundTouchHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        String response = "";
+        checkForReInit();
+        switch (channelUID.getId()) {
+            case BoseSoundTouchBindingConstants.CHANNEL_POWER:
+                if (command instanceof OnOffType) {
+                    response = wsHelper.pressAndReleaseButtonOnSpeaker(BSTKeys.POWER);
+                }
+                break;
+            case BoseSoundTouchBindingConstants.CHANNEL_VOLUME:
+                if (command instanceof RefreshType) {
+                    response = wsHelper.get(REST_SERVICE_VOLUME);
+                    if (response.equals(ERROR_STRING)) {
+                        break;
+                    }
+                    String volume = xmlHelper.parsePath(XmlHelper.REST_CURRENT_VOLUME, response);
+                    if (!volume.isEmpty()) {
+                        updateState(BoseSoundTouchBindingConstants.CHANNEL_VOLUME, new PercentType(volume));
+                    }
+                }
+                if (command instanceof PercentType) {
+                    PercentType num = (PercentType) command;
+                    response = wsHelper.setVolume(num);
+                }
+                break;
+            case BoseSoundTouchBindingConstants.CHANNEL_BASS:
+                if (command instanceof RefreshType) {
+                    response = getBassAndUpdateUI();
+                    if (response.equals(ERROR_STRING)) {
+                        break;
+                    }
+                }
+                if (command instanceof DecimalType) {
+                    DecimalType num = (DecimalType) command;
+                    response = wsHelper.setBass(num);
+                }
+                break;
+            case BoseSoundTouchBindingConstants.CHANNEL_CONTROL:
+                if (command instanceof PlayPauseType) {
+                    PlayPauseType playPause = (PlayPauseType) command;
+                    switch (playPause) {
+                        case PLAY:
+                            response = wsHelper.pressAndReleaseButtonOnSpeaker(BSTKeys.PLAY);
+                            break;
+                        case PAUSE:
+                            response = wsHelper.pressAndReleaseButtonOnSpeaker(BSTKeys.PAUSE);
+                            break;
+                        default:
+                            break;
+                    }
+                }
+            case BoseSoundTouchBindingConstants.CHANNEL_SOURCE:
+                if (command instanceof RefreshType) {
```

Comment:
This body is too big to single case of a switch, please make it a dedicated method.